perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -1899,6 +1899,37 @@ public Application getApplicationById(PerunSession sess, int appId) throws Regis
 
 	}
 
+	@Override
+	public List<Application> getApplicationsForVo(PerunSession userSession, Vo vo, List<String> state, String dateFrom, String dateTo) throws PerunException {
+		vosManager.checkVoExists(userSession, vo);
+
+		//Authorization
+		if (!AuthzResolver.authorizedInternal(userSession, "getApplicationsForVo_Vo_List<String>_policy", Collections.singletonList(vo))) {
+			throw new PrivilegeException(userSession, "getApplicationsForVo");
+		}
```

Comment:
With this change, it will work even if only one bound is specified.
```suggestion
		}
		if (dateTo == null) {
			dateTo = LocalDate.MAX;
		}
		if (dateFrom == null {
			dateFrom = LocalDate.MIN;
		}
```

Reply:
This is extremely ugly from the SQL point of view. I rewrote it.